src/StreetNameRegistry/Municipality/Municipality.cs

```diff
@@ -90,29 +90,19 @@ public void RetireStreetName(PersistentLocalId persistentLocalId)
                 throw new MunicipalityHasInvalidStatusException();
             }
 
-            streetName.Retire();
-        }
-
-        public void CorrectStreetNameName(Names streetNameNames, PersistentLocalId persistentLocalId)
-        {
-            StreetNames
-                .GetNotRemovedByPersistentLocalId(persistentLocalId)
-                .CorrectNames(streetNameNames, GuardStreetNameNames);
+            streetName.CorrectRejection(() => GuardUniqueActiveStreetNameNames(streetName.Names, persistentLocalId));
         }
 
-        private void GuardStreetNameNames(Names streetNameNames, PersistentLocalId persistentLocalId)
+        public void RetireStreetName(PersistentLocalId persistentLocalId)
         {
-            GuardUniqueActiveStreetNameNames(streetNameNames, persistentLocalId);
+            var streetName = StreetNames.GetNotRemovedByPersistentLocalId(persistentLocalId);
 
-            foreach (var streetNameName in streetNameNames)
+            if (MunicipalityStatus != MunicipalityStatus.Current)
```

Comment:
This is a bit bizarre. When a Municipality gets retired all streetnames will retire. So Retire a streetname could not be called, cause all streetnames are retired/rejected.

Reply:
I get your point. It should be considered an idempotant action for streetnames which are already retired because of the municipality being retired. I do think that for a streetname that was rejected because of the municipality being retired, this action shouldn't be allowed.

For this PR, I just moved this code, so that the retire and correct retirement methods stand together. Quickly looked up why the check exists. It falls under GAWR-3030. Maybe we take it up with business?